### ISSUE TITLE
allow app opportunity to handle URI query component upon QURI decoding failure

### ISSUE THREAD

**Issue by thomp** [fukamachi/lack#21]
Should the app have an opportunity to handle the URI query component if QURI fails to decode it?

Currently, by default, a "challenging URI" can result in signalling `babel-encodings:invalid-utf8-starter-byte` (see example below). This might be considered unfortunate as it does not provide an explicit, direct means of establishing where the error occurred, nor does it provide the app an opportunity to handle such queries (e.g., any query where the URI includes a query component which doesn't correspond to UTF-8-encoded character/text data).
#20 would allow the app to handle such situations by leaving the `query-parameters` value empty. This at least permits the app to recognize that the `query-string` value may be in need of TLC.

An alternative approach might be to use a condition (`query-decoding-failure`?) to describe the _specific_ component of the request which proved problematic.

``` lisp
(defparameter *app* (make-instance 'ningle:<app>))

(defparameter *handler* (clack:clackup
          (lack:builder *app*)
          :server :woo
          :port 8083))

(drakma:http-request "http://127.0.0.1:8083/?%BF%27%22%28" :method :get)
```

**Comment by fukamachi** [comment]
Thank you for reporting.
I suppose it isn't a welcome behaviour, however, I don't think your PR that just ignores the query also isn't intuitive.

I noticed `babel:octets-to-string` takes `:errorp` for preventing the error and replace the character by `#\REPLECEMENT_CHARACTER` instead.

In your case, the issue has been resolved with the latest QURI.
Moreover, I think it would be great if it works equally for URL path, so I'll make some changes for each Clack handlers in a few days. (It's already done for Woo, though)
